### PR TITLE
MSP-4838 Docker frankenphp

### PR DESCRIPTION
**Pull request by marinhekman**
see jira ticket

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -73,58 +63,61 @@ RUN mkdir -p /etc/supervisor.d/
...
-    && rm -rf /tmp/blackfire /tmp/blackfire-probe.tar.gz
-
-# Install Blackfire CLI
-RUN mkdir -p /tmp/blackfire \
-    && architecture=$(uname -m) \
-    && curl -A "Docker" -L https://blackfire.io/api/v1/releases/cli/linux/$architecture | tar zxp -C /tmp/blackfire \
-    && mv /tmp/blackfire/blackfire /usr/bin/blackfire \
-    && rm -Rf /tmp/blackfire
+
+## Install Blackfire Probe
```
Comment (marinhekman) [inline Dockerfile:68]: did not work, so disabled for now, Only needed for profiling

File: Dockerfile [hunk 2]
```diff
@@ -73,58 +63,61 @@ RUN mkdir -p /etc/supervisor.d/
...
+#    && mv /tmp/blackfire/blackfire-*.so $(php -r "echo ini_get ('extension_dir');")/blackfire.so \
+#    && printf "extension=blackfire.so\nblackfire.agent_socket=tcp://blackfire:8307\n" > $PHP_INI_DIR/conf.d/blackfire.ini \
+#    && rm -rf /tmp/blackfire /tmp/blackfire-probe.tar.gz
+#
+## Install Blackfire CLI
+#RUN mkdir -p /tmp/blackfire \
+#    && architecture=$(uname -m) \
+#    && curl -A "Docker" -L https://blackfire.io/api/v1/releases/cli/linux/$architecture | tar zxp -C /tmp/blackfire \
+#    && mv /tmp/blackfire/blackfire /usr/bin/blackfire \
+#    && rm -Rf /tmp/blackfire
```
Comment (marinhekman) [inline Dockerfile:83]: did not work, so disabled for now, Only needed for profiling

File: .env [hunk 3]
```diff
@@ -43,7 +43,7 @@ DATABASE_MAX_GAME_SESSION_DBS=99
 ###< doctrine/doctrine-bundle ###
 
 # web server
-# default is "caddy" for docker, for Windows set from db, or derived from URL, e.g. localhost
+# default is "php" for docker, for Windows set from db, or derived from URL, e.g. localhost
```
Comment (marinhekman) [inline .env:46]: so before we had a caddy container, which is gone now... caddy and php are in the "php" container

File: ServerManager/classes/GameSave.php [hunk 4]
```diff
@@ -178,7 +178,7 @@ public function createZip(GameSession $gameSession): bool
     {
         $urlBase = ServerManager::getInstance()->getAbsoluteUrlBase();
         if (getenv('DOCKER') !== false) {
-            $urlBase = 'http://caddy:80'.ServerManager::getInstance()->getAbsolutePathBase();
+            $urlBase = 'http://localhost:80'.ServerManager::getInstance()->getAbsolutePathBase();
```
Comment (marinhekman) [inline ServerManager/classes/GameSave.php:181]: so before from the "php" container we had to request towards the caddy container using http://caddy, But now we can just use "localhost"

File: api/v1/GameSession.php [hunk 5]
```diff
@@ -748,7 +748,7 @@ private function LocalApiRequest(string $apiUrl, int $sessionId, array $postValu
             UrlGeneratorInterface::ABSOLUTE_URL
         );
 
-        $requestHeader = apache_request_headers();
+        $requestHeader = \getallheaders();
```
Comment (marinhekman) [inline api/v1/GameSession.php:751]: somehow apache_request_headers is not available from frankenphp,  but getallheaders is , and that is just an alias

File: docker-compose.yml [hunk 6]
```diff
@@ -77,16 +67,34 @@ services:
       URL_WS_SERVER_SCHEME: ${URL_WS_SERVER_SCHEME:-ws://}
       URL_WS_SERVER_HOST: ${URL_WS_SERVER_HOST:-}
       URL_WS_SERVER_PORT: ${URL_WS_SERVER_PORT:-${WS_SERVER_PORT:-45001}}
-      URL_WS_SERVER_URI: ${URL_WS_SERVER_URI:-}
+      URL_WS_SERVER_URI: ${URL_WS_SERVER_URI:-/ws/}
```
Comment (marinhekman) [inline docker-compose.yml:70]: same fix as in 
https://github.com/BredaUniversityResearch/MSPChallenge-Server/pull/193/files

File: Dockerfile [hunk 7]
```diff
@@ -1,27 +1,24 @@
...
 # https://docs.docker.com/compose/compose-file/#target
 
 
-# Base PHP image
-FROM php_upstream AS php_base
+# Base FrankenPHP image
+FROM frankenphp_upstream AS frankenphp_base
 
-WORKDIR /srv/app
+WORKDIR /app
```
Comment (marinhekman) [inline Dockerfile:16]: note that we go from /srv/app/ to /app/

File: frankenphp/Caddyfile [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+
+	php_server
+
+    @websocket {
+        path /ws/*
+    }
+
+    reverse_proxy @websocket http://php:45001 {
+        # Additional reverse proxy options if needed
+    }
```
Comment (marinhekman) [inline frankenphp/Caddyfile:60]: this was added . Before we had a "route /ws/" block but that did not work anymore

File: frankenphp/conf.d/app.ini [hunk 9]
```diff
@@ -13,7 +13,5 @@ opcache.max_accelerated_files = 20000
 opcache.memory_consumption = 256
 opcache.enable_file_override = 1
 
-; MSP, added "E" = Environment variable parsing
-variables_order = "EGPCS"
-
-implicit_flush=Off
```
Comment (marinhekman) [inline frankenphp/conf.d/app.ini:19]: don't think we need this

File: frankenphp/docker-entrypoint.sh [hunk 10]
```diff
@@ -1,14 +1,20 @@
...
+	setfacl -dR -m u:www-data:rwX -m u:"$(whoami)":rwX composer.json
+	setfacl -R -m u:www-data:rwX -m u:"$(whoami)":rwX composer.lock
+	setfacl -dR -m u:www-data:rwX -m u:"$(whoami)":rwX composer.lock
+	setfacl -R -m u:www-data:rwX -m u:"$(whoami)":rwX symfony.lock
+	setfacl -dR -m u:www-data:rwX -m u:"$(whoami)":rwX symfony.lock
 
-  echo Running $1...
+	#if [ -z "$(ls -A 'vendor/' 2>/dev/null)" ]; then
+	#	composer install --prefer-dist --no-progress --no-interaction
+	#fi
```
Comment (marinhekman) [inline frankenphp/docker-entrypoint.sh:16]: copied this from Symfony docker, might be interesting to only call "bash install.sh" if the vendor folder does not exist ... not sure yet

**Review by JacopoFabrini — APPROVED** [review]

**Pull request merged**